.github/workflows/deploy.yml

```
@@ -1,167 +1,51 @@
-            curl -X POST "https://api.render.com/v1/services/$RENDER_SERVICE_ID/deploys" \
-              -H "Authorization: Bearer $RENDER_TOKEN" \
-              -H "Content-Type: application/json" \
-              -d '{"clearCache": "do_not_clear"}'
-          else
-            echo "Render deployment skipped - missing secrets"
-          fi
-        continue-on-error: true
-
-  # Frontend Testing & Deployment
-  frontend:
+    - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)